
<h1 style="text-align: center"> 
ปัจจัยที่ส่งผลต่อความสนใจเป็นผู้ประกอบการ: กรณีศึกษานักศึกษาชั้นปีที่ 4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการ มหาวิทยาลัยสงขลานครินทร์
</h1>

<h3 style="text-align: center"> 
The Factors Influencing to be an Entrepreneurs: Case Study of Fourth-Year Students, Faculty of Economics and Management Sciences, Prince of Songkla University
</h3>

<p style="text-align: center"> 
การศึกษานี้เป็นส่วนหนึ่งของวิชา Research Project in Economics (875 - 417) ปีการศึกษา 2564 สาขาเศรษฐศาสตร์ คณะเศรษฐศาสตร์ มหาวิทยาลัยสงขลานครินทร์
</p>


<p style="text-align: right"> 
<strong>ภัคพล ตันติวิชช์</strong>
</p>

## ภาพรวม

&emsp; การศึกษาเรื่องปัจจัยที่ส่งผลต่อความสนใจเป็นผู้ประกอบการ: กรณีศึกษานักศึกษาชั้นปีที่ 4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการ มหาวิทยาลัยสงขลานครินทร์ มีวัตถุประสงค์เพื่อศึกษาถึงปัจจัยที่ส่งผลต่อความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจของนักศึกษาชั้นปีที่ 4 และศึกษาถึงความแตกต่างของปัจจัยส่วนบุคคลและคุณลักษณะส่วนบุคคล ระหว่างผู้ที่สนใจเป็นผู้ประกอบการวิสากิจกับบุคคลที่ไม่สนใจเป็นผู้ประกอบการวิสาหกิจ โดยใช้ทฤษฎีพฤติกรรมตามแผน (Theory of Planned Behavior) ทฤษฎีบุคลิกภาพ 5 องค์ประกอบ (five-factor model of personality) และทฤษฎีคุณลักษณะของการเป็นผู้ประกอบการ (Entrepreneurial Characteristics) ซึ่งเป็นการวิจัยเชิงปริมาณ (Quantitative Research) ใช้ข้อมูลปฐมภูมิ (Primary Data) โดยมีแบบสอบถาม (Questionnaire) เป็นเครื่องมือในการเก็บรวบรวมข้อมูล มีประชากรคือนักศึกษาชั้นปีที่ 4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการ มหาวิทยาลัยสงขลานครินทร์ วิทยาเขตหาดใหญ่ จังหวัดสงขลา จำนวน 734 คน และได้ใช้กลุ่มตัวอย่างในการศึกษาครั้งนี้จำนวนทั้งสิ้น 148 คน มากกว่าขนาดตัวอย่างที่คำนวณได้จากสูตร โดยทำการสุ่มกลุ่มตัวอย่างด้วยวิธีการสุ่มกลุ่มตัวอย่างอย่างง่าย (Simple Random Sampling) จากนั้นใช้การวิเคราะห์ข้อมูลด้วยสถิติเชิงพรรณนา (Descriptive statistics) เพื่ออธิบาย บรรยาย และสรุปลักษณะทั่วไปของกลุ่มข้อมูลที่ได้เก็บรวบรวมมา ด้วยวิธีแจกแจงความถี่ (Frequency) ซึ่งแสดงผลเป็นค่าร้อยละ (Percentage) และใช้การวิเคราะห์ข้อมูลด้วยสถิติเชิงอนุมาน (Inferential Statistics) เพื่อตอบวัตถุประสงค์ของงานวิจัย ได้แก่ การใช้แบบจำลองโลจิท (Logit Model) และวิเคราะห์ข้อมูลด้วยวิธีการวิเคราะห์ความน่าจะเป็นสูงสุด (Maximum Likelihood Estimation : MLE)

## วัตถุประสงค์

- ศึกษาถึงปัจจัยที่ส่งผลต่อความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจของนักศึกษาชั้นปีที่ 4

- ศึกษาถึงความแตกต่างของปัจจัยส่วนบุคคลและคุณลักษณะส่วนบุคคล ระหว่างผู้ที่สนใจเป็นผู้ประกอบการวิสากิจกับบุคคลที่ไม่สนใจเป็นผู้ประกอบการวิสาหกิจ

## กรอบแนวคิด

**กรอบแนวคิดในการวิเคราะห์**

![กรอบแนวคิด.png](กรอบแนวคิด.png)

## วิธีวิจัย

### การเก็บรวบรวมข้อมูล

<div style="text-align: left;">

- ใช้ Google Forms เก็บรวบรวมข้อมูล

</div>

### การเลือกตัวอย่าง

- กำหนดขนาดกลุ่มตัวอย่างจากสูตร : $\mathrm{n}\ = \ \frac{p(1-p)}{\frac{e^2}{z^2}+\frac{p(1-p)}{N}} \ =\  \frac{0.5(0.5)}{\frac{0.1^2}{1.645^2}+\frac{0.5(0.5)}{734}}\ = \ 61.94\ \approx\ 62\ คน$

&emsp;&emsp;&emsp; โดยที่ 

&emsp;&emsp;&emsp; $n$  = ขนาดของกลุ่มตัวอย่าง

&emsp;&emsp;&emsp; $N$ = จำนวนประชากรทั้งหมด

&emsp;&emsp;&emsp; $z$ = ค่า z-score ซึ่ง ± $z$ จะสอดคล้องกับระดับความเชื่อมั่นที่เราต้องการ (กำหนดไว้ที่ 90%)

&emsp;&emsp;&emsp; $p$ = ค่าประมาณของสัดส่วนประชากร (ประมาณการค่าแบบ conservative ไว้ที่ 0.5)

&emsp;&emsp;&emsp; $e$ = ค่าความคลาดเคลื่อนของการสุ่มตัวอย่างที่ยอมรับได้ (กำหนดไว้ที่ 0.1)


- Simple Random Sampling : n = 148

### การวิเคราะห์ข้อมูล

#### **สถิติเชิงพรรณนา (Descriptive statistics)** 

&emsp; เพื่ออธิบาย บรรยาย และสรุปลักษณะทั่วไปของกลุ่มข้อมูลที่ได้เก็บรวบรวมมา ด้วยวิธีแจกแจงความถี่ (Frequency) ซึ่งแสดงผลเป็นค่าร้อยละ (Percentage) แต่มีข้อจำกัดคือจะไม่สามารถอ้างอิงลักษณะประชากรได้

#### **สถิติเชิงอนุมาน (Inferential Statistics)**

##### **Logistic Regression** : 
&emsp; หาปัจจัยที่ส่งผลต่อความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจ ในกลุ่มนักศึกษาชั้นปีที่ 4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการมหาวิทยาลัยสงขลานครินทร์ วิทยาเขตหาดใหญ่ จังหวัดสงขลา ด้วยแบบจำลองโลจิท (Logit Model) และวิเคราะห์ข้อมูลด้วยวิธีการวิเคราะห์ความน่าจะเป็นสูงสุด (Maximum Likelihood Estimation : MLE) แสดงในรูปสมการได้ดังนี้

$$
Prob(y = 1|x_1 x_2 … x_{22}) = G(β_0+β_1x_1 + ...+β_{22}x_{22})
$$

$$
Prob(y = 1|x_1 x_2 … x_{22}) = \frac{e^{β_0+β_1x_1 + ...+β_{22}x_{22}}} {1+e^{β_0+β_1x_1 + ...+β_{22}x_{22}}}
$$

โดยที่ 

&emsp; $y$ คือ ความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจ ; มีค่า = 1 ถ้าสนใจ , 0 ถ้าไม่สนใจ

&emsp; $x_1$ คือ เพศ ; มีค่า 1 ถ้าเป็นเพศชาย , 0 ถ้าเป็นเพศหญิง

&emsp; $x_2$ คือ คณะที่ศึกษา ; มีค่า 1 ถ้าศึกษาคณะเศรษฐศาสตร์ , 0 ถ้าศึกษาคณะวิทยาการจัดการ

&emsp; $x_3$ คือ ผลการเรียนเฉลี่ย

&emsp; $x_4$ คือ อาชีพของบิดา-มารดา ; มีค่า 1 ถ้าพ่อหรือแม่ประกอบธุรกิจส่วนตัว , 0 ถ้าไม่ใช่

&emsp; $x_5$ คือ รายได้เฉลี่ยของครอบครัวต่อเดือน ; มีค่า 1 ถ้ารายได้เฉลี่ย 20,001 – 40,000 บาท , 0 ถ้าไม่ใช่

&emsp; $x_6$ คือ รายได้เฉลี่ยของครอบครัวต่อเดือน ; มีค่า 1 ถ้ารายได้เฉลี่ย 40,001 – 60,000 บาท , 0 ถ้าไม่ใช่

&emsp; $x_7$ คือ รายได้เฉลี่ยของครอบครัวต่อเดือน ; มีค่า 1 ถ้ารายได้เฉลี่ย 60,001 – 80,000 บาท , 0 ถ้าไม่ใช่

&emsp; $x_8$ คือ รายได้เฉลี่ยของครอบครัวต่อเดือน ; มีค่า 1 ถ้ารายได้เฉลี่ย 80,001 – 100,000 บาท, 0 ถ้าไม่ใช่

&emsp; $x_9$ คือ รายได้เฉลี่ยของครอบครัวต่อเดือน ; มีค่า 1 ถ้ารายได้เฉลี่ยมากกว่า 100,000 บาท , 0 ถ้าไม่ใช่

&emsp; $x_{10}$ คือ ค่าคะแนนเฉลี่ยทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ

&emsp; $x_{11}$ คือ ค่าคะแนนเฉลี่ยบรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง

&emsp; $x_{12}$ คือ ค่าคะแนนเฉลี่ยการรับรู้ถึงความสามารถในการควยคุมพฤติกรรม

&emsp; $x_{13}$ คือ ค่าคะแนนเฉลี่ยบุคลิกภาพแบบมั่นคงทางอารมณ์

&emsp; $x_{14}$ คือ ค่าคะแนนเฉลี่ยบุคลิกภาพแบบแสดงตัว

&emsp; $x_{15}$ คือ ค่าคะแนนเฉลี่ยบุคลิกภาพแบบเปิดรับประสบการณ์

&emsp; $x_{16}$ คือ ค่าคะแนนเฉลี่ยบุคลิกภาพแบบเห็นด้วยกับผู้อื่น

&emsp; $x_{17}$ คือ ค่าคะแนนเฉลี่ยบุคลิกภาพแบบมีจิตสำนึก

&emsp; $x_{18}$ คือ ค่าคะแนนเฉลี่ยคุณลักษณะด้านความเป็นอิสระ

&emsp; $x_{19}$ คือ ค่าคะแนนเฉลี่ยคุณลักษณะด้านการสร้างนวัตกรรม

&emsp; $x_{20}$ คือ ค่าคะแนนเฉลี่ยคุณลักษณะด้านการทำงานเชิงลึก

&emsp; $x_{21}$ คือ ค่าคะแนนเฉลี่ยคุณลักษณะด้านความสามารถในการแข่งขัน

&emsp; $x_{22}$ คือ ค่าคะแนนเฉลี่ยคุณลักษณะด้านการเผชิญกับความเสี่ยง

##### T - test :
&emsp; หาความแตกต่างของปัจจัยส่วนบุคคลและคุณลักษณะส่วนบุคคล ได้แก่ ผลการเรียนเฉลี่ย (GPAX) ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง การรับรู้ถึงความสามรถในการควบคุมพฤติกรรม บุคลิกภาพแบบมั่นคงทางอารมณ์ บุคลิกภาพแบบแสดงตัว บุคลิกภาพแบบเปิดรับประสบการณ์ บุคลิกภาพแบบเห็นด้วยกับผู้อื่น บุคลิกภาพแบบมีจิตสำนึก คุณลักษณะด้านความเป็นอิสระ คุณลักษณะด้านการสร้างนวัตกรรม คุณลักษณะด้านการทำงานเชิงลึก คุณลักษณะด้านความสามารถในการแข่งขัน และคุณลักษณะด้านการเผชิญกับความเสี่ยง ของบุคคลระหว่างผู้ที่สนใจเป็นผู้ประกอบการวิสากิจกับบุคคลที่ไม่สนใจเป็นผู้ประกอบการวิสาหกิจ โดยใช้สูตร T-test for Independent sample ที่ระดับความเชื่อมั่น 95% (α = 0.05) ดังนี้

- ในกรณีที่ความแปรปรวน (variance) ของกลุ่มตัวอย่างเท่ากัน $*s_1^2=s_2^2*$ ใช้สูตร

$$
t\ = \ \frac{\bar{x}_1-\bar{x}_2}{\sqrt{\frac{(n_1-1)\cdot s_1^2 + (n_2-1)\cdot s_2^2}{n_1+n_2-2}\cdot(\frac{1}{n_1}+\frac{1}{n_2})}}
$$

และ $$df  =  n_1+n_2 − 2$$

โดยที่

&emsp; t คือ ค่าสถิติ T-test

&emsp; $\bar{x}_1$  และ $\bar{x}_2$   คือ ค่าเฉลี่ยของกลุ่มตัวอย่างที่ 1 และ 2 ตามลำดับ

&emsp; $s_1^2$ และ $s_2^2$    คือ ค่าความแปรปรวน (variance) ของกลุ่มตัวอย่างที่ 1 และ 2 ตามลำดับ

&emsp; $n_1$ และ $n_2$   คือ ขนาดของกลุ่มตัวอย่างที่ 1 และ 2 ตามลำดับ

&emsp; $df$               คือ ระดับความเป็นอิสระ

## ผลการศึกษา

In [3]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import statsmodels.formula.api as smf
from statsmodels.stats.weightstats import ttest_ind



set this to be able to render plotly in VScode

In [4]:
from plotly.offline import plot, iplot, init_notebook_mode
init_notebook_mode(connected=True)

import responses from Google Forms

In [5]:
url = open("url.txt", "r").readline()
df = pd.read_csv(url, encoding="utf-8")

df["ท่านสนใจจะประกอบกิจการอะไรในอนาคต "].fillna("ไม่สนใจ" , inplace=True)

### สถิติเชิงพรรณนา (Descriptive statistics)

#### ปัจจัยส่วนบุคคล

##### เพศ

In [31]:
fig1 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["เพศ"].value_counts().keys()), 
                              values=df["เพศ"].value_counts(),
                              title="เพศ"
                              )])
fig1.show()

- เพศชายจำนวน 85 คน คิดเป็นร้อยละ 57.43
- เพศหญิงจำนวน 63 คน คิดเป็นร้อยละ 42.57

##### คณะที่ศึกษา

In [7]:
fig2 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["คณะที่ศึกษา"].value_counts().keys()), 
                              values=df["คณะที่ศึกษา"].value_counts(),
                              title="คณะที่ศึกษา"
                            )])
fig2.show()

- คณะวิทยาการจัดการจำนวน 90 คน คิดเป็นร้อยละ 60.81
- คณะเศรษฐศาสตร์จำนวน 58 คน คิดเป็นร้อยละ 39.19

##### สาขาที่ศึกษา

In [8]:
fig3 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["สาขาที่ศึกษา"].value_counts().keys()), 
                              values=df["สาขาที่ศึกษา"].value_counts(),
                              title="สาขาที่ศึกษา"
                            )])
fig3.show()

- สาขาเศรษฐศาสตร์จำนวน 36 คน คิดเป็นร้อยละ 24.32
- สาขาเศรษฐศาสตร์เกษตรและธุรกิจเกษตร 22 คน คิดเป็นร้อยละ 14.86
- สาขาการเงินและการลงทุนจำนวน 17 คน คิดเป็นร้อยละ 11.49
- สาขารัฐประศาสนศาสตรจำนวน 17 คน คิดเป็นร้อยละ 11.49
- สาขาการตลาดจำนวน 16 คน คิดเป็นร้อยละ 10.81
- สาขาการจัดการทรัพยากรมนุษย์จำนวน 12 คน คิดเป็นร้อยละ 8.11
- สาขาการจัดการโลจิสติกส์จำนวน 10 คน คิดเป็นร้อยละ 6.76
- สาขาการจัดการไมซ์จำนวน 7 คน คิดเป็นร้อยละ 4.73
- สาขาระบบสารสนเทศทางธุรกิจจำนวน 5 คน คิดเป็นร้อยละ 3.38
- สาขาบัญชีจำนวน 4 คน คิดเป็นร้อยละ 2.7
- สาขาการจัดการ(ภาษาอังกฤษ)จำนวน 2 คน คิดเป็นร้อยละ 1.35

##### อาชีพของบิดา

In [9]:
fig4 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["อาชีพของบิดา"].value_counts().keys()), 
                              values=df["อาชีพของบิดา"].value_counts(),
                              title="อาชีพของบิดา"
                            )])
fig4.show()

- ธุรกิจส่วนตัวจำนวน 54 คน คิดเป็นร้อยละ 36.49
- ข้าราชการจำนวน 29 คน คิดเป็นร้อยละ 19.59
- พนักงานเอกชนจำนวน 22 คน คิดเป็นร้อยละ 14.86
- รัฐวิสาหกิจจำนวน 19 คน คิดเป็นร้อยละ 12.84
- เกษียณจำนวน 13 คน คิดเป็นร้อยละ 8.78
- อื่น ๆ (เกษตรกร , ว่างาน , ถึงแก่กรรม) จำนวน 11 คน คิดเป็นร้อยละ 7.43

##### อาชีพของมารดา

In [10]:
fig5 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["อาชีพของมารดา"].value_counts().keys()), 
                              values=df["อาชีพของมารดา"].value_counts(),
                              title="อาชีพของมารดา"
                            )])
fig5.show()

- ธุรกิจส่วนตัวจำนวน 53 คน คิดเป็นร้อยละ 35.81
- ข้าราชการจำนวน 28 คน คิดเป็นร้อยละ 18.92
- รัฐวิสาหกิจจำนวน 21 คน คิดเป็นร้อยละ 14.19
- พนักงานเอกชนจำนวน 20 คน คิดเป็นร้อยละ 13.51
- เกษียณจำนวน 13 คน คิดเป็นร้อยละ 8.78
- อื่น ๆ (เกษตรกร , ว่างาน) จำนวน 13 คน คิดเป็นร้อยละ 8.78

##### รายได้เฉลี่ยของครอบครัวต่อเดือน

In [11]:
fig6 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["รายได้เฉลี่ยของครอบครัวต่อเดือน"].value_counts().keys()), 
                              values=df["รายได้เฉลี่ยของครอบครัวต่อเดือน"].value_counts(),
                              title="รายได้เฉลี่ยของครอบครัวต่อเดือน"
                            )])
fig6.show()

- รายได้ไม่เกิน 20,000 บาท จำนวน 13 คน คิดเป็นร้อยละ 8.78
- รายได้ 20,001 – 40,000 บาท จำนวน 30 คน คิดเป็นร้อยละ 20.27
- รายได้ 40,001 – 60,000 บาท จำนวน 43 คน คิดเป็นร้อยละ 29.05
- รายได้ 60,001 – 80,000 บาท จำนวน 31 คน คิดเป็นร้อยละ 20.95
- รายได้ 80,001 – 100,000 บาท จำนวน 16 คน คิดเป็นร้อยละ 10.81
- รายได้มากกว่า 100,000 บาท จำนวน 15 คน คิดเป็นร้อยละ 10.1

#### ปัจจัยทฤษฎีพฤติกรรมตามแผน

In [12]:
group_mean = df.groupby("ท่านมีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจในอนาคตหรือไม่").mean()

#ปัจจัยทฤษฎีพฤติกรรมตามแผน
xlabels1 = ['ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ','บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง',"การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม"]
TPB_tot = [df["TPB1"].mean(), df["TPB2"].mean(), df["TPB3"].mean()]
TPB_g0 = [group_mean["TPB1"][0], group_mean["TPB2"][0], group_mean["TPB3"][0]]
TPB_g1 = [group_mean["TPB1"][1], group_mean["TPB2"][1], group_mean["TPB3"][1]]

fig7 = go.Figure()
fig7.add_trace(go.Bar(
    x=xlabels1,
    y=TPB_tot,
    name='ทั้งหมด'))
fig7.add_trace(go.Bar(
    x=xlabels1,
    y=TPB_g0,
    name='กลุ่มที่สนใจ'))
fig7.add_trace(go.Bar(
    x=xlabels1,
    y=TPB_g1,
    name='กลุ่มที่ไม่สนใจ'))

fig7.update_layout(
    title='ปัจจัยทฤษฎีพฤติกรรมตามแผน',
    xaxis_tickfont_size=14,
    yaxis=dict(
        title='คะแนนเฉลี่ย',
        titlefont_size=16,
        tickfont_size=14
    ),
    barmode='group', xaxis_tickangle=45)

fig7.show()

- กลุ่มผู้ตอบแบบสอบถามทั้งหมด 148 คน
    - ค่าเฉลี่ยทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ เท่ากับ 4.19
    - ค่าเฉลี่ยบรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง เท่ากับ 4.2
    - ค่าเฉลี่ยการรับรู้ถึงความสามารถในการควบคุมพฤติกรรม เท่ากับ 4.12

- กลุ่มที่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ เท่ากับ 4.24
    - ค่าเฉลี่ยบรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง เท่ากับ 4.21
    - ค่าเฉลี่ยการรับรู้ถึงความสามารถในการควบคุมพฤติกรรม เท่ากับ 4.22

- กลุ่มที่ไม่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ เท่ากับ 4.09
    - ค่าเฉลี่ยบรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง เท่ากับ 4.19
    - ค่าเฉลี่ยการรับรู้ถึงความสามารถในการควบคุมพฤติกรรม เท่ากับ 3.92

#### ปัจจัยบุคลิกภาพ 5 องค์ประกอบ

In [13]:
xlabels2 = ["บุคลิกภาพแบบมั่นคงทางอารมณ์", "บุคลิกภาพแบบแสดงตัว", "บุคลิกภาพแบบเปิดรับประสบการณ์", "บุคลิกภาพแบบเห็นด้วยกับผู้อื่น", "บุคลิกภาพแบบมีจิตสํานึก"]
FFM_tot = [df["FFM1"].mean(), df["FFM2"].mean(), df["FFM3"].mean(), df["FFM4"].mean(), df["FFM5"].mean()]
FFM_g0 = [group_mean["FFM1"][0], group_mean["FFM2"][0], group_mean["FFM3"][0], group_mean["FFM4"][0], group_mean["FFM5"][0]]
FFM_g1 = [group_mean["FFM1"][1], group_mean["FFM2"][1], group_mean["FFM3"][1], group_mean["FFM4"][1], group_mean["FFM5"][1]]

fig8 = go.Figure()
fig8.add_trace(go.Bar(
    x=xlabels2,
    y=FFM_tot,
    name='ทั้งหมด'))
fig8.add_trace(go.Bar(
    x=xlabels2,
    y=FFM_g0,
    name='กลุ่มที่สนใจ'))
fig8.add_trace(go.Bar(
    x=xlabels2,
    y=FFM_g1,
    name='กลุ่มที่ไม่สนใจ'))

fig8.update_layout(
    title='ปัจจัยบุคลิกภาพ 5 องค์ประกอบ',
    xaxis_tickfont_size=14,
    yaxis=dict(
        title='คะแนนเฉลี่ย',
        titlefont_size=16,
        tickfont_size=14
    ),
    barmode='group', xaxis_tickangle=45)

fig8.show()

- กลุ่มผู้ตอบแบบสอบถามทั้งหมด 148 คน
    - ค่าเฉลี่ยบุคลิกภาพแบบมั่นคงทางอารมณ์ เท่ากับ 3.07
    - ค่าเฉลี่ยบุคลิกภาพแบบแสดงตัว เท่ากับ 3.67
    - ค่าเฉลี่ยบุคลิกภาพแบบเปิดรับประสบการณ์ เท่ากับ 3.82
    - ค่าเฉลี่ยบุคลิกภาพแบบเห็นด้วยกับผู้อื่น เท่ากับ 3.84
    - ค่าเฉลี่ยบุคลิกภาพแบบมีจิตสำนึก เท่ากับ 3.90

- กลุ่มที่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยบุคลิกภาพแบบมั่นคงทางอารมณ์ เท่ากับ 3.00
    - ค่าเฉลี่ยบุคลิกภาพแบบแสดงตัว เท่ากับ 4.02
    - ค่าเฉลี่ยบุคลิกภาพแบบเปิดรับประสบการณ์ เท่ากับ 3.09
    - ค่าเฉลี่ยบุคลิกภาพแบบเห็นด้วยกับผู้อื่น เท่ากับ 3.96
    - ค่าเฉลี่ยบุคลิกภาพแบบมีจิตสำนึก เท่ากับ 3.96

- กลุ่มที่ไม่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยบุคลิกภาพแบบมั่นคงทางอารมณ์ เท่ากับ 3.20
    - ค่าเฉลี่ยบุคลิกภาพแบบแสดงตัว เท่ากับ 3.01
    - ค่าเฉลี่ยบุคลิกภาพแบบเปิดรับประสบการณ์ เท่ากับ 3.30
    - ค่าเฉลี่ยบุคลิกภาพแบบเห็นด้วยกับผู้อื่น เท่ากับ 3.63
    - ค่าเฉลี่ยบุคลิกภาพแบบมีจิตสำนึก เท่ากับ 3.79

#### ปัจจัยคุณลักษณะความเป็นผู้ประกอบการ

In [14]:
xlabels3 = ["คุณลักษณะด้านความเป็นอิสระ", "คุณลักษณะด้านการสร้างนวัตกรรม", "คุณลักษณะด้านการทํางานเชิงลึก", "คุณลักษณะด้านความสามารถในการแข่งขัน", "คุณลักษณะด้านการเผชิญกับความเสี่ยง"]
EC_tot = [df["EC1"].mean(), df["EC2"].mean(), df["EC3"].mean(), df["EC4"].mean(), df["EC5"].mean()]
EC_g0 = [group_mean["EC1"][0], group_mean["EC2"][0], group_mean["EC3"][0], group_mean["EC4"][0], group_mean["EC5"][0]]
EC_g1 = [group_mean["EC1"][1], group_mean["EC2"][1], group_mean["EC3"][1], group_mean["EC4"][1], group_mean["EC5"][1]]

fig9 = go.Figure()
fig9.add_trace(go.Bar(
    x=xlabels3,
    y=EC_tot,
    name='ทั้งหมด'))
fig9.add_trace(go.Bar(
    x=xlabels3,
    y=EC_g0,
    name='กลุ่มที่สนใจ'))
fig9.add_trace(go.Bar(
    x=xlabels3,
    y=EC_g1,
    name='กลุ่มที่ไม่สนใจ'))

fig9.update_layout(
    title='ปัจจัยคุณลักษณะความเป็นผู้ประกอบการ',
    xaxis_tickfont_size=14,
    yaxis=dict(
        title='คะแนนเฉลี่ย',
        titlefont_size=16,
        tickfont_size=14
    ),
    barmode='group', xaxis_tickangle=45)

fig9.show()

- กลุ่มผู้ตอบแบบสอบถามทั้งหมด 148 คน
    - ค่าเฉลี่ยคุณลักษณะด้านความเป็นอิสระ เท่ากับ 4.01
    - ค่าเฉลี่ยคุณลักษณะด้านการสร้างนวัตกรรม เท่ากับ 3.84
    - ค่าเฉลี่ยคุณลักษณะด้านการทำงานเชิงลึก เท่ากับ 3.98
    - ค่าเฉลี่ยสำหรับคุณลักษณะด้านความสามารถในการแข่งขัน เท่ากับ 3.91
    - ค่าเฉลี่ยคุณลักษณะด้านการเผชิญกับความเสี่ยง เท่ากับ 4.00

- กลุ่มที่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยคุณลักษณะด้านความเป็นอิสระ เท่ากับ 4.06
    - ค่าเฉลี่ยคุณลักษณะด้านการสร้างนวัตกรรม เท่ากับ 4.20
    - ค่าเฉลี่ยคุณลักษณะด้านการทำงานเชิงลึก เท่ากับ 4.16
    - ค่าเฉลี่ยสำหรับคุณลักษณะด้านความสามารถในการแข่งขัน เท่ากับ 3.99
    - ค่าเฉลี่ยคุณลักษณะด้านการเผชิญกับความเสี่ยง เท่ากับ 4.19

- กลุ่มที่ไม่สนใจเป็นผู้ประกอบการ
    - ค่าเฉลี่ยคุณลักษณะด้านความเป็นอิสระ เท่ากับ 3.93
    - ค่าเฉลี่ยคุณลักษณะด้านการสร้างนวัตกรรม เท่ากับ 3.16
    - ค่าเฉลี่ยคุณลักษณะด้านการทำงานเชิงลึก เท่ากับ 3.65
    - ค่าเฉลี่ยสำหรับคุณลักษณะด้านความสามารถในการแข่งขัน เท่ากับ 3.75
    - ค่าเฉลี่ยคุณลักษณะด้านการเผชิญกับความเสี่ยง เท่ากับ 3.62

#### ความสนใจเป็นผู้ประกอบการวิสาหกิจ

ผลการศึกษาในส่วนนี้ ได้ทำการสอบถามว่าในอนาคตผู้ตอบแบบสอบถามสนใจเป็นผู้ประกอบการหรือไม่ ซึ่งเป็นคำถามให้ผู้ตอบแบบสอบถามตอบตามความสนใจของตนเอง โดยไม่ได้คำนึงถึงสถานการณ์โควิด-19 ที่เกิดขึ้นในช่วงเวลาในปัจจุบัน เนื่องจากต้องการดูความสนใจที่มีอยู่ในปัจจุบันอย่างแท้จริงว่าในอนาคตผู้ตอบแบบสอบถามสนใจอยากเป็นผู้ประกอบการหรือไม่

In [15]:
fig10 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["ท่านมีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจในอนาคตหรือไม่"].value_counts().keys()), 
                              values=df["ท่านมีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจในอนาคตหรือไม่"].value_counts(),
                              title="ความสนใจเป็นผู้ประกอบการวิสาหกิจ"
                            )])
fig10.show()

- สนใจที่จะเป็นผู้ประกอบการวิสาหกิจ จำนวน 97 คน คิดเป็นร้อยละ 65.54
- ไม่มีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจ จำนวน 51 คน คิดเป็นร้อยละ 34.46

##### กิจการที่สนใจ

In [16]:
fig11 = go.Figure(data=[go.Pie(labels=pd.DataFrame(df["ท่านสนใจจะประกอบกิจการอะไรในอนาคต "].value_counts().keys()), 
                              values=df["ท่านสนใจจะประกอบกิจการอะไรในอนาคต "].value_counts(),
                              title="กิจการที่สนใจ"
                            )])
fig11.show()

- ไม่มีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจ จำนวน 51 คน คิดเป็นร้อยละ 34.46
- การผลิตอาหารและเครื่องดื่ม จำนวน 26 คน คิดเป็นร้อยละ 17.57
- ร้านอาหาร/ภัตตาคาร จำนวน 17 คน คิดเป็นร้อยละ 11.49
- กิจกรรมอสังหาริมทรัพย์ จำนวน 16 คน คิดเป็นร้อยละ 10.81
- บริการขนส่งสินค้า จำนวน 6 คน คิดเป็นร้อยละ 4.05
- โรงแรม/เกสต์เฮ้าส์/บังกะโล จำนวน 6 คน คิดเป็นร้อยละ 4.05
- การผลิตเสื้อผ้าและสิ่งทอ จำนวน 5 คน คิดเป็นร้อยละ 3.38
- การท่องเที่ยว จำนวน 4 คน คิดเป็นร้อยละ 2.70
- บริการเสริมความงาม จำนวน 4 คน คิดเป็นร้อยละ 2.70
- การผลิตไม้และเฟอร์นิเจอร์ จำนวน 4 คน คิดเป็นร้อยละ 2.70
- การค้าวัสดุก่อสร้าง จำนวน 4 คน คิดเป็นร้อยละ 2.70
- การค้าสินค้าเกษตร จำนวน 3 คน คิดเป็นร้อยละ 2.03
- ร้านค้าของชำ จำนวน 1 คน คิดเป็นร้อยละ 0.68
- อื่น ๆ (การจัดการขยะอินทรีย์) จำนวน 1 คน คิดเป็นร้อยละ 0.68

### สถิติเชิงอนุมาน (Inferential Statistics)

#### Logistic Regression

##### Logit Model

In [17]:
model = "to_be_entrepreneur ~ gender + faculty + gpax + parentsjob + inc2 + inc3 + inc4 + inc5 + inc6 + TPB1 + TPB2 + TPB3 + FFM1 + FFM2 + FFM3 + FFM4 + FFM5 + EC1 + EC2 + EC3 + EC4 + EC5"
logit_model = smf.logit(formula=model , data=df).fit()

Optimization terminated successfully.
         Current function value: 0.161395
         Iterations 10


**Result**

In [18]:
print(
    f"chi-square : {logit_model.llr} \n"
    f"sig. level of chi-square test : {logit_model.llr_pvalue} \n"
    f"pseudo-rsquared : {logit_model.prsquared} \n"
    f"Log-Likelihood : {logit_model.llf} \n"
    )

chi-square : 142.86182000736733 
sig. level of chi-square test : 1.0506054251532209e-19 
pseudo-rsquared : 0.7494009668644621 
Log-Likelihood : -23.886434331425498 



get coefficient

In [19]:
df_result = pd.DataFrame(logit_model.summary2().tables[1],columns=['Coef.', 'Std.Err.', 'z', 'P>|z|', '[0.025', '0.975]']).drop(columns = ['z','[0.025', '0.975]'])

get marginal effect

In [20]:
margeff = logit_model.get_margeff().summary().tables[1].as_html()
df_margeff = pd.read_html(margeff, header=0, index_col=0)[0].drop(columns=['std err', 'z', 'P>|z|', '[0.025', '0.975]'])
df_margeff

dy/dx
gender      0.1370
faculty     0.0616
gpax        0.0980
parentsjob  0.0836
inc2       -0.2584
inc3       -0.2658
inc4       -0.2118
inc5       -0.3353
inc6       -0.3476
TPB1       -0.0151
TPB2       -0.0082
TPB3       -0.1252
FFM1       -0.0351
FFM2        0.1600
FFM3        0.0729
FFM4        0.0951
FFM5        0.0025
EC1        -0.0403
EC2         0.1682
EC3         0.1840
EC4        -0.0554
EC5         0.0732

Merge DataFrame

In [21]:
df_result["dy/dx"] = df_margeff["dy/dx"]
df_result = df_result[["Coef.", "dy/dx", "Std.Err.", "P>|z|"]]
df_result

Coef.   dy/dx   Std.Err.     P>|z|
Intercept  -37.270108     NaN  13.803433  0.006933
gender       2.664399  0.1370   1.144964  0.019962
faculty      1.196982  0.0616   1.047259  0.253052
gpax         1.904821  0.0980   1.294289  0.141099
parentsjob   1.625957  0.0836   1.047462  0.120595
inc2        -5.024360 -0.2584   2.270432  0.026901
inc3        -5.168658 -0.2658   2.322279  0.026035
inc4        -4.118511 -0.2118   1.969058  0.036473
inc5        -6.519403 -0.3353   2.549331  0.010549
inc6        -6.759061 -0.3476   2.834839  0.017112
TPB1        -0.294020 -0.0151   1.988390  0.882447
TPB2        -0.158988 -0.0082   0.936174  0.865146
TPB3        -2.434942 -0.1252   1.390892  0.080009
FFM1        -0.682513 -0.0351   0.704511  0.332658
FFM2         3.109998  0.1600   1.391825  0.025452
FFM3         1.417156  0.0729   0.917387  0.122401
FFM4         1.849984  0.0951   1.183438  0.117999
FFM5         0.048495  0.0025   1.026290  0.962312
EC1         -0.784386 -0.0403   1.437155  0.585210
EC2          3.269840  0.1682   1.332593  0.014138
EC3          3.578249  0.1840   1.248255  0.004149
EC4         -1.077279 -0.0554   1.524248  0.479715
EC5          1.422885  0.0732   0.794208  0.073201

rename columns

In [22]:
new_col_names =  {'Intercept' : 'ค่าคงที่',
                  'gender' : "เพศ",
                  'faculty' : 'คณะที่ศึกษา',
                  'gpax' : 'ผลการเรียนเฉลี่ย (GPAX)',
                  'parentsjob' :'อาชีพของบิดา-มารดา',
                  'inc2' : "รายได้เฉลี่ย 20000-40000",
                  'inc3' : "รายได้เฉลี่ย 40000-60000",
                  'inc4' : "รายได้เฉลี่ย 60000-80000",
                  'inc5' : "รายได้เฉลี่ย 80000-100000",
                  'inc6' : "รายได้เฉลี่ย >100000",
                  'TPB1' : 'ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ',
                  'TPB2' : 'บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง',
                  'TPB3' : "การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม",
                  'FFM1' : "บุคลิกภาพแบบมั่นคงทางอารมณ์",
                  'FFM2' : "บุคลิกภาพแบบแสดงตัว",
                  'FFM3' : "บุคลิกภาพแบบเปิดรับประสบการณ์",
                  'FFM4' : "บุคลิกภาพแบบเห็นด้วยกับผู้อื่น",
                  'FFM5' : "บุคลิกภาพแบบมีจิตสํานึก",
                  'EC1' : "คุณลักษณะด้านความเป็นอิสระ",
                  'EC2' : "คุณลักษณะด้านการสร้างนวัตกรรม",
                  'EC3' : "คุณลักษณะด้านการทํางานเชิงลึก",
                  'EC4' : "คุณลักษณะด้านความสามารถในการแข่งขัน",
                  'EC5' : "คุณลักษณะด้านการเผชิญกับความเสี่ยง",
                  'to_be_entrepreneur' : "ความสนใจเป็นผู้ประกอบการ"}

df_result.rename(index = new_col_names, inplace=True)

In [23]:
df_result

Coef.   dy/dx   Std.Err.  \
ค่าคงที่                                     -37.270108     NaN  13.803433   
เพศ                                            2.664399  0.1370   1.144964   
คณะที่ศึกษา                                    1.196982  0.0616   1.047259   
ผลการเรียนเฉลี่ย (GPAX)                        1.904821  0.0980   1.294289   
อาชีพของบิดา-มารดา                             1.625957  0.0836   1.047462   
รายได้เฉลี่ย 20000-40000                      -5.024360 -0.2584   2.270432   
รายได้เฉลี่ย 40000-60000                      -5.168658 -0.2658   2.322279   
รายได้เฉลี่ย 60000-80000                      -4.118511 -0.2118   1.969058   
รายได้เฉลี่ย 80000-100000                     -6.519403 -0.3353   2.549331   
รายได้เฉลี่ย >100000                          -6.759061 -0.3476   2.834839   
ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ        -0.294020 -0.0151   1.988390   
บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง  -0.158988 -0.0082   0.936174   
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม     -2.434942 -0.1252   1.390892   
บุคลิกภาพแบบมั่นคงทางอารมณ์                   -0.682513 -0.0351   0.704511   
บุคลิกภาพแบบแสดงตัว                            3.109998  0.1600   1.391825   
บุคลิกภาพแบบเปิดรับประสบการณ์                  1.417156  0.0729   0.917387   
บุคลิกภาพแบบเห็นด้วยกับผู้อื่น                 1.849984  0.0951   1.183438   
บุคลิกภาพแบบมีจิตสํานึก                        0.048495  0.0025   1.026290   
คุณลักษณะด้านความเป็นอิสระ                    -0.784386 -0.0403   1.437155   
คุณลักษณะด้านการสร้างนวัตกรรม                  3.269840  0.1682   1.332593   
คุณลักษณะด้านการทํางานเชิงลึก                  3.578249  0.1840   1.248255   
คุณลักษณะด้านความสามารถในการแข่งขัน           -1.077279 -0.0554   1.524248   
คุณลักษณะด้านการเผชิญกับความเสี่ยง             1.422885  0.0732   0.794208   

                                                 P>|z|  
ค่าคงที่                                      0.006933  
เพศ                                           0.019962  
คณะที่ศึกษา                                   0.253052  
ผลการเรียนเฉลี่ย (GPAX)                       0.141099  
อาชีพของบิดา-มารดา                            0.120595  
รายได้เฉลี่ย 20000-40000                      0.026901  
รายได้เฉลี่ย 40000-60000                      0.026035  
รายได้เฉลี่ย 60000-80000                      0.036473  
รายได้เฉลี่ย 80000-100000                     0.010549  
รายได้เฉลี่ย >100000                          0.017112  
ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ        0.882447  
บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง  0.865146  
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม     0.080009  
บุคลิกภาพแบบมั่นคงทางอารมณ์                   0.332658  
บุคลิกภาพแบบแสดงตัว                           0.025452  
บุคลิกภาพแบบเปิดรับประสบการณ์                 0.122401  
บุคลิกภาพแบบเห็นด้วยกับผู้อื่น                0.117999  
บุคลิกภาพแบบมีจิตสํานึก                       0.962312  
คุณลักษณะด้านความเป็นอิสระ                    0.585210  
คุณลักษณะด้านการสร้างนวัตกรรม                 0.014138  
คุณลักษณะด้านการทํางานเชิงลึก                 0.004149  
คุณลักษณะด้านความสามารถในการแข่งขัน           0.479715  
คุณลักษณะด้านการเผชิญกับความเสี่ยง            0.073201

Logistic Regression : List variables that Significant at 0.1 level

In [24]:
df_result_sig = df_result.loc[df_result["P>|z|"]<0.1]
df_result_sig

Coef.   dy/dx   Std.Err.  \
ค่าคงที่                                  -37.270108     NaN  13.803433   
เพศ                                         2.664399  0.1370   1.144964   
รายได้เฉลี่ย 20000-40000                   -5.024360 -0.2584   2.270432   
รายได้เฉลี่ย 40000-60000                   -5.168658 -0.2658   2.322279   
รายได้เฉลี่ย 60000-80000                   -4.118511 -0.2118   1.969058   
รายได้เฉลี่ย 80000-100000                  -6.519403 -0.3353   2.549331   
รายได้เฉลี่ย >100000                       -6.759061 -0.3476   2.834839   
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม  -2.434942 -0.1252   1.390892   
บุคลิกภาพแบบแสดงตัว                         3.109998  0.1600   1.391825   
คุณลักษณะด้านการสร้างนวัตกรรม               3.269840  0.1682   1.332593   
คุณลักษณะด้านการทํางานเชิงลึก               3.578249  0.1840   1.248255   
คุณลักษณะด้านการเผชิญกับความเสี่ยง          1.422885  0.0732   0.794208   

                                              P>|z|  
ค่าคงที่                                   0.006933  
เพศ                                        0.019962  
รายได้เฉลี่ย 20000-40000                   0.026901  
รายได้เฉลี่ย 40000-60000                   0.026035  
รายได้เฉลี่ย 60000-80000                   0.036473  
รายได้เฉลี่ย 80000-100000                  0.010549  
รายได้เฉลี่ย >100000                       0.017112  
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม  0.080009  
บุคลิกภาพแบบแสดงตัว                        0.025452  
คุณลักษณะด้านการสร้างนวัตกรรม              0.014138  
คุณลักษณะด้านการทํางานเชิงลึก              0.004149  
คุณลักษณะด้านการเผชิญกับความเสี่ยง         0.073201

#### T - test

In [25]:
df_to_be_etpn = df[df["ท่านมีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจในอนาคตหรือไม่"]=="สนใจ"]
df_not_to_be_etpn = df[df["ท่านมีความสนใจที่จะเป็นผู้ประกอบการวิสาหกิจในอนาคตหรือไม่"]=="ไม่สนใจ"]

In [26]:
def ttest_score(x, y) :
    cols_ttest = ["gpax", "TPB1", "TPB2", "TPB3", "FFM1", "FFM2", "FFM3", "FFM4", "FFM5", "EC1", "EC2", "EC3", "EC4", "EC5"]
    new_col_names =  {
                  'gpax' : 'ผลการเรียนเฉลี่ย (GPAX)',
                  'TPB1' : 'ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ',
                  'TPB2' : 'บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง',
                  'TPB3' : "การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม",
                  'FFM1' : "บุคลิกภาพแบบมั่นคงทางอารมณ์",
                  'FFM2' : "บุคลิกภาพแบบแสดงตัว",
                  'FFM3' : "บุคลิกภาพแบบเปิดรับประสบการณ์",
                  'FFM4' : "บุคลิกภาพแบบเห็นด้วยกับผู้อื่น",
                  'FFM5' : "บุคลิกภาพแบบมีจิตสํานึก",
                  'EC1' : "คุณลักษณะด้านความเป็นอิสระ",
                  'EC2' : "คุณลักษณะด้านการสร้างนวัตกรรม",
                  'EC3' : "คุณลักษณะด้านการทํางานเชิงลึก",
                  'EC4' : "คุณลักษณะด้านความสามารถในการแข่งขัน",
                  'EC5' : "คุณลักษณะด้านการเผชิญกับความเสี่ยง"}
    ttest_score = []
    p_value_score = []
    degree_of_freedom = []

    for col in cols_ttest :
        test = ttest_ind(x[col], y[col])
        ttest_score.append(test[0])
        p_value_score.append(test[1])
        degree_of_freedom.append(test[2])
    df = pd.DataFrame({
        "Variable" : cols_ttest,
        "T-test" : ttest_score,
        "P-Value" : p_value_score
    }).set_index("Variable")
    df.rename(index = new_col_names, inplace=True)
    return df

In [27]:
df_ttest = ttest_score(df_to_be_etpn, df_not_to_be_etpn)
df_ttest


T-test       P-Value
Variable                                                             
ผลการเรียนเฉลี่ย (GPAX)                        3.343303  1.051930e-03
ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ         2.164427  3.205725e-02
บรรทัดฐานทางสังคมหรือการคล้อยตามกลุ่มอ้างอิง   0.231126  8.175407e-01
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม      3.010328  3.075624e-03
บุคลิกภาพแบบมั่นคงทางอารมณ์                   -1.739055  8.413295e-02
บุคลิกภาพแบบแสดงตัว                           10.260071  6.188265e-19
บุคลิกภาพแบบเปิดรับประสบการณ์                  7.425169  8.630861e-12
บุคลิกภาพแบบเห็นด้วยกับผู้อื่น                 3.388821  9.026723e-04
บุคลิกภาพแบบมีจิตสํานึก                        1.794005  7.488178e-02
คุณลักษณะด้านความเป็นอิสระ                     1.851845  6.606781e-02
คุณลักษณะด้านการสร้างนวัตกรรม                  9.528751  4.907471e-17
คุณลักษณะด้านการทํางานเชิงลึก                  6.742656  3.371020e-10
คุณลักษณะด้านความสามารถในการแข่งขัน            3.627797  3.944384e-04
คุณลักษณะด้านการเผชิญกับความเสี่ยง             5.458278  2.012716e-07

T-test : List variable Significant at 0.05 level

In [28]:
df_ttest_sig = df_ttest[df_ttest["P-Value"] < 0.05]
df_ttest_sig

T-test       P-Value
Variable                                                          
ผลการเรียนเฉลี่ย (GPAX)                     3.343303  1.051930e-03
ทัศนะคติต่อการเป็นผู้ประกอบการวิสาหกิจ      2.164427  3.205725e-02
การรับรู้ถึงความสามารถในการควบคุมพฤติกรรม   3.010328  3.075624e-03
บุคลิกภาพแบบแสดงตัว                        10.260071  6.188265e-19
บุคลิกภาพแบบเปิดรับประสบการณ์               7.425169  8.630861e-12
บุคลิกภาพแบบเห็นด้วยกับผู้อื่น              3.388821  9.026723e-04
คุณลักษณะด้านการสร้างนวัตกรรม               9.528751  4.907471e-17
คุณลักษณะด้านการทํางานเชิงลึก               6.742656  3.371020e-10
คุณลักษณะด้านความสามารถในการแข่งขัน         3.627797  3.944384e-04
คุณลักษณะด้านการเผชิญกับความเสี่ยง          5.458278  2.012716e-07

## สรุปผลการศึกษา

1. นักศึกษาชั้นปีที่4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการ ส่วนใหญ่สนใจประกอบกิจการการผลิตอาหารและเครื่องดื่ม รองลงมาเป็นการบริการร้านอาหาร และกิจกรรมอสังหาริมทรัพย์ตามลําดับ

2. นักศึกษาเพศชาย มีโอกาสสนใจเป็นผู้ประกอบการมากกว่าเพศหญิง 13.70 %
    - ผลการศึกษาในส่วนนี้ยังสอดคล้องกับผลการศึกษาที่สรุปว่า “นักศึกษาที่กล้าตัดสินใจ ยอมรับความเสี่ยงที่อาจสร้างผลประโยชน์ในอนาคต จะยิ่งมีโอกาสที่จะสนใจเป็นผู้ประกอบการเพิ่มขึ้น” เนื่องด้วยเพศชายจะมีบุคลิกภาพที่ชื่นชอบและยอมรับความเสี่ยงได้มากกว่าเพศหญิง

3. นักศึกษาที่ครอบครัวมีรายได้เฉลี่ยต่อเดือนตํ่ากว่า 20,000 บาท จะมีความสนใจเป็นผู้ประกอบการมากกว่าครอบครัวที่มีรายได้เฉลี่ยสูงกว่า 20,000 บาท
    - เหตุผลหนึ่งอาจเป็นฐานะทางบ้านที่ไม่ดีจะมีส่วนกระตุ้นให้ตัวนักศึกษามีความกระตือรือร้นที่จะยกระดับฐานะความเป็นอยู่ให้ดีขึ้น ซึ่งหนึ่งในทางเลือกนั่นก็คือการเป็นผู้ประกอบการ

4. ความพร้อมในด้านการเงิน ฯลฯ จะลดโอกาสที่จะสนใจที่เป็นผู้ประกอบการลง
    - สอดคล้องกับข้อสรุปผลการศึกษารายได้เฉลี่ยของครอบครัวต่อเดือนที่สรุปว่า “ครอบครัวที่มีฐานะดีหรือมีความพร้อมทางด้านการเงิน (รายได้เฉลี่ยสูงกว่า 20,000 บาทต่อเดือน) จะมีโอกาสสนใจเป็นผู้ประกอบการน้อยกว่าครอบครัวที่มีฐานะไม่ดี (รายได้เฉลี่ยต่ำกว่า 20,000 บาทต่อเดือน) ”

5. นักศึกษาที่ชื่นชอบการเข้าสังคม ชอบทํากิจกรรม ใช้ชีวิตอย่างสนุกสนาน ตื่นเต้น มองโลกในแง่ดี จะยิ่งมีโอกาสที่จะสนใจที่เป็นผู้ประกอบการเพิ่มขึ้น

6. นักศึกษาที่มีความพยายามนําเสนอสิ่งใหม่ ๆ เพื่อการพัฒนา จะยิ่งมีโอกาสที่จะสนใจที่เป็นผู้ประกอบการเพิ่มขึ้น

7. นักศึกษาที่มองการไกล แสวงหาโอกาสความเป็นไปได้ในอนาคต  จะยิ่งมีโอกาสที่จะสนใจที่เป็นผู้ประกอบการเพิ่มขึ้น

8. นักศึกษาที่กล้าตัดสินใจ ยอมรับความเสี่ยงที่อาจสร้างผลประโยชน์ในอนาคต จะยิ่งมีโอกาสที่จะสนใจที่เป็นผู้ประกอบการเพิ่มขึ้น

9. ผลการเรียนเฉลี่ย (GPAX) ทัศนะคติต่อการเป็น ผู้ประกอบการวิสาหกิจ การรับรู้ถึงความสามรถในการควบคุมพฤติกรรม บุคลิกภาพแบบแสดงตัว บุคลิกภาพแบบ เปิดรับประสบการณ์ บุคลิกภาพแบบเห็นด้วยกับผู้อื่น คุณลักษณะด้านการสร้างนวัตกรรม คุณลักษณะด้านการ ทํางานเชิงลึก คุณลักษณะด้านความสามารถในการแข่งขัน และคุณลักษณะด้านการเผชิญกับความเสี่ยง ระหว่างผู้ที่สนใจเป็นผู้ประกอบการ และผู้ที่ไม่สนใจเป็นผู้ประกอบการ **มีความแตกต่างกันอย่างมีนัยสําคัญทางสถิติ**ที่ระดับความเชื่อมั่น 95% (α = 0.05)

## ข้อเสนอแนะ

เพื่อผลการศึกษาที่ดีขึ้นในอนาคตสำหรับการศึกษาเรื่องปัจจัยที่ส่งผลต่อความสนใจเป็นผู้ประกอบการ ผู้วิจัยมีข้อเสนอแนะเพื่อการศึกษาเพิ่มเติมในอนาคต ดังนี้

1. การศึกษาครั้งนี้เป็นเก็บข้อมูลเฉพาะกลุ่มตัวอย่างที่เป็นนักศึกษาชั้นปีที่ 4 คณะเศรษฐศาสตร์และคณะวิทยาการจัดการ มหาวิทยาลัยสงขลานครินทร์เท่านั้น ในการศึกษาครั้งต่อไปอาจขยายขอบเขตคณะที่ทำการศึกษาให้หลากหลายขึ้น หรือทั้งมหาวิทยาลัย เพื่อให้เห็นถึงความแตกต่างในมิติสาขาที่เรียนได้ชัดเจนและหลากหลายยิ่งขึ้น

2. การศึกษาครั้งนี้เป็นการวิจัยเชิงปริมาณ (Quantitative Research) ใช้ข้อมูลปฐมภูมิ (Primary Data) โดยมีแบบสอบถาม (Questionnaire) เป็นเครื่องมือในการเก็บรวบรวมข้อมูล ดังนั้นการศึกษา ครั้งต่อไป อาจใช้การวิจัยเชิงคุณภาพร่วมด้วย เช่นการสัมภาษณ์เชิงลึก หรือการสัมภาษณ์กลุ่ม เพื่อให้เข้าใจถึงกระบวนการอย่างละเอียดของปัจจัยต่าง ๆ ที่ส่งผลต่อความสนใจที่จะเป็นผู้ประกอบการ และช่วยค้นหาปัจจัยใหม่ที่ทำให้เกิดความสนใจที่จะเป็นผู้ประกอบการ